xbmc/addons/AddonInfo.h

```diff
@@ -85,13 +85,240 @@ namespace ADDON
   typedef std::map<std::string, std::string> InfoMap;
   typedef std::map<std::string, std::string> ArtMap;
 
+  class CAddonBuilder;
+
   class CAddonInfo
   {
   public:
     CAddonInfo();
     CAddonInfo(std::string id, TYPE type);
 
     /*!
+     * @brief To ask generated class is usable and all needed parts are set.
+     *
+     * @return True is OK and usable, otherwise false.
+     *
+     * @note This function is only be related to constructor's who read the
+     * addon xml part.
+     */
+    bool IsUsable() const { return m_usable; }
```

Comment:
Can remove it now, but becomes on next commits used the know class construction with read of addon.xml was OK.